time.go

```diff
@@ -118,13 +118,19 @@ func (t Time) IsBankingDay() bool {
 	return true
 }
 
-// AddBankingDay takes an integer for the number of valid banking days to add and returns a Time
+// AddBankingDay takes an integer for the number of valid banking days to add and returns a Time.
+// Negative values and large values (over 500 days) will not modify the Time.
 func (t Time) AddBankingDay(d int) Time {
-	t.Time = t.Time.AddDate(0, 0, d)
-	if !t.IsBankingDay() {
-		return t.AddBankingDay(1)
+	if d < 1 || d > 500 {
```

Comment:
Looks odd to check for? Looks good otherwise.

Reply:
Adding negative banking days doesn't seem like a valid use-case. The 500+ guard is just to prevent huge recursive calls. IIRC Go doesn't perform tail call optimization so passing large values could OOM the service.

Reply:
will it be better to raise an error instead of returning if its out of bounds ?

Reply:
Yea, but that would require changing and breaking the method signature.

Reply:
Nothing on my side then, it would be nice if you could tag a paygate version after this merge